pyxem/tests/test_signals/test_diffraction_variance2d.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+import numpy as np
+
+from pyxem.signals.diffraction_variance2d import DiffractionVariance2D
+from pyxem.signals.diffraction_variance2d import ImageVariance
+from pyxem.signals.diffraction_variance1d import DiffractionVariance1D
+from pyxem.signals.electron_diffraction2d import ElectronDiffraction2D
+
+
+class TestDiffractionVariance:
+
+    def test_get_diffraction_variance_signal(self,
+                                             diffraction_pattern):
+        difvar = DiffractionVariance2D(diffraction_pattern)
+        assert isinstance(difvar, DiffractionVariance2D)
+
+    def test_get_dif_var_radial_profile(self,
+                                   diffraction_pattern):
+        difvar = DiffractionVariance2D(diffraction_pattern)
+        rp = difvar.get_radial_profile()
+        assert isinstance(rp, DiffractionVariance1D)
+
+
+    @pytest.fixture
+    def axes_test_dp(self):
+        dp_data = np.random.randint(0, 10, (2, 2, 10, 10))
+        dp = ElectronDiffraction2D(dp_data)
+        return dp
+
+    def test_radial_profile_axes(self, axes_test_dp):
```

Comment:
this could be a lot more readable, a `for` loop might help, but also declaring, `n_scale`, `name` and `units` all on the top line

Reply:
For loop doesn't work since I'm intentionally naming the different axes different things. Have moved stuff up.

Reply:
Snippet
```
for x in ([1,2],[2,3],[3,4]):
...     print(x[0])
1
2
3
```
So you would have something like
```

for axis_assignment in ([0,n_scale,name,units],[1,n_scale*2,units,name]):
     axes_test_dpetcetc.scale[axis_assignment[0]] = axis_assignments[1] #always n scale
     etc.name[axis_assignment[0]] = axis_assignments[2] # name in 0 case, units in case 1
     etc.units[axis_assignments[0] = axis_assigments[3]   # units in 0 case, name in case 1
```

obviously with slightly better naming, but it saves a lot of lines